core/store/src/main/java/uk/gov/gchq/gaffer/store/operation/handler/PathHandler.java

```diff
@@ -0,0 +1,175 @@
+ * limitations under the License.
+ */
+
+package uk.gov.gchq.gaffer.store.operation.handler;
+
+import com.google.common.collect.HashBasedTable;
+import com.google.common.collect.ImmutableSet;
+import com.google.common.collect.Iterables;
+import com.google.common.collect.Lists;
+import com.google.common.collect.Sets;
+
+import uk.gov.gchq.gaffer.commonutil.stream.Streams;
+import uk.gov.gchq.gaffer.data.element.Edge;
+import uk.gov.gchq.gaffer.data.element.Element;
+import uk.gov.gchq.gaffer.operation.OperationChain;
+import uk.gov.gchq.gaffer.operation.OperationException;
+import uk.gov.gchq.gaffer.operation.data.EntitySeed;
+import uk.gov.gchq.gaffer.operation.impl.Path;
+import uk.gov.gchq.gaffer.operation.impl.get.GetElements;
+import uk.gov.gchq.gaffer.operation.impl.output.ToEntitySeeds;
+import uk.gov.gchq.gaffer.operation.impl.output.ToSet;
+import uk.gov.gchq.gaffer.operation.impl.output.ToVertices;
+import uk.gov.gchq.gaffer.store.Context;
+import uk.gov.gchq.gaffer.store.Store;
+
+import java.util.ArrayList;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Set;
+import java.util.Stack;
+import java.util.stream.Collectors;
+
+public class PathHandler implements OutputOperationHandler<Path, Iterable<Iterable<Edge>>> {
+
+    private final Set<EntitySeed> visitedSeeds = new HashSet<>();
+    private int pathVertices;
+    private int hops;
+
+    @Override
+    public Iterable<Iterable<Edge>> doOperation(final Path operation, final Context context, final Store store) throws OperationException {
+
+        // Check input
+        if (null == operation.getInput() || !operation.getInput().iterator().hasNext()) {
+            return null;
+        }
+
+        // Check operations input
+        final Iterable<GetElements> operations = operation.getOperations();
+
+        if (null == operations || Iterables.isEmpty(operations)) {
+            return null;
+        }
+
+        // Validate the View objects
+        for (final GetElements op : operations) {
+            if (null != op.getView() && op.getView().hasEntities()) {
+                throw new OperationException("The view for operation " + op + " must not contain Entities.");
+            }
+        }
+
+        Iterable<? extends EntitySeed> seeds = operation.getInput();
+
+        hops = Iterables.size(operation.getOperations());
+        pathVertices = hops + 1;
+
+        final HashBasedTable<Object, Object, Set<Edge>> edgeTable = HashBasedTable.create();
+
+        // Execute the GetElements operations
+        for (final GetElements op : operations) {
+
+            final Set<EntitySeed> opSeeds = difference(visitedSeeds, Sets.newHashSet(seeds));
+
+            op.setInput(opSeeds);
+            final Iterable<? extends Element> results = store.execute(op, context);
+            visitedSeeds.addAll(opSeeds);
+
+            // Cache results
+            Lists.newArrayList(results).stream()
```

Comment:
I think we should avoid doing: Lists.newArrayList(results)
As it will bring all the results into memory. Instead can we do: Streams.toStream(results)

Reply:
Actually you could just do this logic in a for loop instead of using streams.